homeassistant/components/roborock/image.py

```diff
@@ -98,13 +98,16 @@ def is_map_valid(self) -> bool:
             and bool(self.coordinator.roborock_device_info.props.status.in_cleaning)
         )
 
-    def _handle_coordinator_update(self):
+    def _handle_coordinator_update(self) -> None:
         # Bump last updated every third time the coordinator runs, so that async_image
         # will be called and we will evaluate on the new coordinator data if we should
         # update the cache.
         if (
-            dt_util.utcnow() - self.image_last_updated
-        ).total_seconds() > IMAGE_CACHE_INTERVAL and self.is_map_valid():
+            self.image_last_updated is not None
```

Comment:
Because the property can be possible `None`, but is used to calculate below. This check ensures a `None` value doesn't get into that calculation (as it would be invalid).

In this case, you can argue this fixes a bug or typing caught a bug; or that this was a case that could never happen but is not cleared up in typing.

Reply:
Exactly. That's why I'm curious. :)

Reply:
I've updated the typing on the class so it is just datetime opposed to datetime | None - does this answer your questions? It should never be none as it is set in init.

Reply:
It does. Thanks!